src/evm_parse.c

```diff
@@ -218,41 +289,27 @@ enum parse_rv parse_rlp_txn(struct EVM_RLP_list_state *const state, evm_parser_m
       case 2: { // Now parse items.
           uint8_t itemStartIdx;
           switch(state->item_index) {
-#define PARSE_ITEM(ITEM, save) \
-            case ITEM: {\
-                itemStartIdx = meta->input.consumed; \
-                PRINTF("Entering " #ITEM "\n");                          \
-                sub_rv = parse_rlp_item ## save(state, meta); \
-                PRINTF("Exiting " #ITEM "\n");                          \
-                state->remaining -= meta->input.consumed - itemStartIdx; \
-            } (void)0
-#define FINISH_ITEM_CHUNK() \
-            if(sub_rv != PARSE_RV_DONE) return sub_rv;                  \
-            state->item_index++;                                        \
-            init_rlp_item(&state->rlpItem_state);
 
-            PARSE_ITEM(EVM_TXN_NONCE, );
+            PARSE_ITEM(EVM_LEGACY_TXN_NONCE, );
             FINISH_ITEM_CHUNK();
-            // Don't need to do anything in particular with the nonce.
-            // In particular, all values are at least plausible here. We could show it perhaps.
 
-            PARSE_ITEM(EVM_TXN_GASPRICE, _to_buffer);
+            PARSE_ITEM(EVM_LEGACY_TXN_GASPRICE, _to_buffer);
             uint64_t gasPrice = enforceParsedScalarFits64Bits(&state->rlpItem_state);
-            size_t gasPriceLength = state->rlpItem_state.length;
-            state->gasPrice = gasPrice;
+            size_t gasPriceLength = state->rlpItem_state.length; // TODO is this length handling dead code? now that enforceParsed handles length
+            state->priorityFeePerGas = gasPrice;
             FINISH_ITEM_CHUNK();
 
-            PARSE_ITEM(EVM_TXN_STARTGAS, _to_buffer);
+            PARSE_ITEM(EVM_LEGACY_TXN_STARTGAS, _to_buffer);
             uint64_t startGas = enforceParsedScalarFits64Bits(&state->rlpItem_state);
-            size_t startGasLength = state->rlpItem_state.length;
-            state->startGas = startGas;
+            size_t startGasLength = state->rlpItem_state.length; // TODO is this length handling dead code? now that enforceParsed handles length
+            state->gasLimit = startGas;
             // TODO: We don't currently support the C-chain gas limit of 100 million,
             // which would have a fee larger than what fits in a word
             if(gasPriceLength + startGasLength > 8)
```

Comment:
The length handling might not be dead code yet as the product of the two things can still overflow 64bits (this check is actually more conservative, as noted below)
This can probably be replaced now that there's 256 bit support by converting the two terms to 256 before multiplying. 
Getting rid of the intermediate lengths would also allow sharing the code between `PARSE_ITEM` and `FINISH_ITEM_CHUNK` between legacy and non-legacy transaction parsers.
If that ends up not being trivial, let's leave that as TODO instead of the others added above.

Reply:
TODO added :+1: